### ISSUE TITLE
Shibboleth: Convert account page fails due to password validation.

### ISSUE THREAD

**Issue by kcondon** [IQSS/dataverse#3045]
While testing account conversion found it would not accept a valid password.

**Comment by pdurbin** [comment]
Huh. Good catch @kcondon. I just tried on dataverse-internal (after first verifying the username and password are good by logging in via non-Shib) and got this:

![shib_-_2016-03-30_11 23 46](https://cloud.githubusercontent.com/assets/21006/14147783/e22e9480-f66a-11e5-9fae-f5802f3d888d.png)


```
[2016-03-30T11:22:23.828-0400] [glassfish 4.1] [INFO] [] [edu.harvard.iq.dataverse.Shib] [tid: _ThreadID=52 _ThreadName=jk-connector(3)] [timeMillis: 1459351343828] [levelValue: 800] [[
  Debug summary: getting username from the builtin user we looked up via email (state: PROMPT_TO_CONVERT_EXISTING_ACCOUNT).]]

[2016-03-30T11:23:09.338-0400] [glassfish 4.1] [INFO] [] [edu.harvard.iq.dataverse.Shib] [tid: _ThreadID=53 _ThreadName=jk-connector(4)] [timeMillis: 1459351389338] [levelValue: 800] [[
  builtin username: kc32816]]

[2016-03-30T11:23:09.339-0400] [glassfish 4.1] [INFO] [] [edu.harvard.iq.dataverse.authorization.providers.shib.ShibServiceBean] [tid: _ThreadID=53 _ThreadName=jk-connector(4)] [timeMillis: 1459351389339] [levelValue: 800] [[
  checking to see if kc32816 knows the password...]]

[2016-03-30T11:23:09.340-0400] [glassfish 4.1] [INFO] [] [edu.harvard.iq.dataverse.authorization.providers.shib.ShibServiceBean] [tid: _ThreadID=53 _ThreadName=jk-connector(4)] [timeMillis: 1459351389340] [levelValue: 800] [[
  The username and/or password you entered is invalid. Need assistance accessing your account?Bad username, email address, or password]]
```

This was working recently since @eaquigley was able to grab screenshots for #2975.

Soon, I'll try this on my laptop and on shibtest.

**Comment by pdurbin** [comment]
I just verified this bug on https://shibtest.dataverse.org running v. 4.3 build 4-origin/2939-shib-78ef2c1

I must have introduced a regression while refactoring code to make it more testable. My guess is that my API tests still work but I'll confirm either way once I'm in the code to fix the bug.

**Comment by pdurbin** [comment]
@kcondon as we discussed 643e455 has a bug fix for a regression I introduced at 7915bf5 when I attempted to internationalize some of the strings in BuiltinAuthenticationProvider. Maybe @michbarsinai can advise us on the state of the code after 7915bf5 and how he envisions BuiltinAuthenticationProvider being internationalized. @kcondon suggests that maybe this should go in a separate issue, which isn't a bad idea. It may come up on its own as @scolapasta coordinates internationalization efforts in #209. Or maybe it's all a non-issue. I'm really not sure.

Anyway, the Shib bug should be fixed. Passing to QA.
